src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/validators.py

```diff
@@ -120,29 +126,59 @@ def _validate_network_parameters(namespace, formula=None):
             lab_vnet = lab_vnets[0]
             namespace.vnet_name = lab_vnet.name
             namespace.lab_virtual_network_id = lab_vnet.id
+    # User did provide vnet or has been selected from formula
     else:
-        namespace.lab_virtual_network_id = resource_id(subscription=get_subscription_id(),
-                                                       resource_group=namespace.resource_group,
-                                                       namespace='microsoft.devtestlab',
-                                                       type='labs',
-                                                       name=namespace.lab_name,
-                                                       child_type='virtualnetworks',
-                                                       child_name=namespace.vnet_name)
-
-    # Select default subnet for selected virtual network when subnet is not provided
+        lab_vnet = vnet_operation.get_resource(namespace.resource_group, namespace.lab_name, namespace.vnet_name)
+        namespace.lab_virtual_network_id = lab_vnet.id
+
+    # User did not provide subnet and not selected from formula
     if not namespace.subnet:
-        # Get the first subnet of the lab's virtual network
-        lab_vnet = vnet_operation.get_resource(namespace.resource_group,
-                                               namespace.lab_name,
-                                               namespace.vnet_name)
         namespace.subnet = lab_vnet.subnet_overrides[0].lab_subnet_name
 
-        # Determine value for disallow_public_ip_address based on subnet's
-        # use_public_ip_address_permission property
-        if lab_vnet.subnet_overrides[0].use_public_ip_address_permission == 'Allow':
-            namespace.disallow_public_ip_address = False
-        else:
-            namespace.disallow_public_ip_address = True
+    _validate_ip_configuration(namespace, lab_vnet)
+
+
+# pylint: disable=no-member
```

Comment:
There are few lines that'd require this comment. All the lines that accessed `lab_vnet`, so i'd prefer to keep it on method if that's fine & doesn't have drawbacks from python perspective :)